implementations/rust/ockam/ockam_api/tests/authority.rs

```diff
@@ -0,0 +1,390 @@
+async fn test_one_admin_one_member(ctx: &mut Context) -> Result<()> {
+    use std::collections::HashMap;
+
+    let secure_channels = secure_channels();
+
+    let admins = setup(ctx, secure_channels.clone(), 1).await?;
+    let admin = &admins[0];
+
+    let now = now()?;
+
+    let member = secure_channels
+        .identities()
+        .identities_creation()
+        .create_identity()
+        .await?
+        .identifier()
+        .clone();
+
+    let mut attributes = HashMap::<&str, &str>::default();
+    attributes.insert("key", "value");
+
+    admin.client.add_member(member.clone(), attributes).await?;
+
+    let members = admin.client.list_member_ids().await?;
+    assert_eq!(members.len(), 2);
+    assert!(members.contains(&admin.identifier));
+    assert!(members.contains(&member));
+
+    let members = admin.client.list_members().await?;
+    assert_eq!(members.len(), 2);
+    assert!(members.get(&admin.identifier).is_some());
+
+    let attrs = members.get(&member).unwrap();
+
+    assert_eq!(attrs.attrs().len(), 2);
+    assert_eq!(
+        attrs.attrs().get("trust_context_id".as_bytes()),
+        Some(&b"123456".to_vec())
+    );
+    assert_eq!(
+        attrs.attrs().get("key".as_bytes()),
+        Some(&b"value".to_vec())
+    );
+
+    assert!(attrs.added() - now < 5.into());
+    assert!(attrs.expires().is_none());
+    assert_eq!(attrs.attested_by(), Some(admin.identifier.clone()));
+
+    admin.client.delete_member(member.clone()).await?;
+
+    let members = admin.client.list_member_ids().await?;
+    assert_eq!(members.len(), 1);
+    assert_eq!(members[0], admin.identifier);
+
+    let members = admin.client.list_members().await?;
+    assert_eq!(members.len(), 1);
+    assert!(members.get(&admin.identifier).is_some());
+
+    ctx.stop().await?;
+
+    Ok(())
+}
+
+#[ockam_macros::test]
+async fn test_two_admins_two_members(ctx: &mut Context) -> Result<()> {
+    use std::collections::HashMap;
+
+    let secure_channels = secure_channels();
+
+    let admins = setup(ctx, secure_channels.clone(), 2).await?;
+    let admin1 = &admins[0];
+    let admin2 = &admins[1];
+
+    let member1 = secure_channels
+        .identities()
+        .identities_creation()
+        .create_identity()
+        .await?
+        .identifier()
+        .clone();
+    let member2 = secure_channels
+        .identities()
+        .identities_creation()
+        .create_identity()
+        .await?
+        .identifier()
+        .clone();
+
+    let mut attributes1 = HashMap::<&str, &str>::default();
+    attributes1.insert("key1", "value1");
+
+    let mut attributes2 = HashMap::<&str, &str>::default();
+    attributes2.insert("key2", "value2");
+
+    let now = now()?;
+
+    admin1
+        .client
+        .add_member(member1.clone(), attributes1)
+        .await?;
+
+    admin2
+        .client
+        .add_member(member2.clone(), attributes2)
+        .await?;
+
+    let mut members1 = admin1.client.list_member_ids().await?;
```

Comment:
The important point that should underlined here is that each admin can see all the members.